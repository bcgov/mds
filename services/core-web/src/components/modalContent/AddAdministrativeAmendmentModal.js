import React from "react";
import PropTypes from "prop-types";
import Highlight from "react-highlighter";
import AdministrativeAmendmentForm from "@/components/Forms/noticeOfWork/AdministrativeAmendmentForm";

const propTypes = {
  onSubmit: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
};

const defaultProps = {};

export const AddAdministrativeAmendmentModal = (props) => (
  <div>
    <p>
      Creating an <Highlight search="Administrative Amendment">Administrative Amendment</Highlight>{" "}
      allows you to update/change the selected permit amendment.
    </p>
    <br />
    <AdministrativeAmendmentForm {...props} />
  </div>
);

AddAdministrativeAmendmentModal.propTypes = propTypes;
AddAdministrativeAmendmentModal.defaultProps = defaultProps;

export default AddAdministrativeAmendmentModal;
