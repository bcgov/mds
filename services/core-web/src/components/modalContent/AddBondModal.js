import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Alert } from "antd";
import {
  getDropdownProvinceOptions,
  getBondTypeDropDownOptions,
  getBondDocumentTypeDropDownOptions,
  getBondDocumentTypeOptionsHash,
} from "@common/selectors/staticContentSelectors";
import BondForm from "@/components/Forms/Securities/BondForm";
import CustomPropTypes from "@/customPropTypes";

const propTypes = {
  onSubmit: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  bondTypeDropDownOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
  provinceOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
  bondDocumentTypeDropDownOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
  bondDocumentTypeOptionsHash: PropTypes.objectOf(PropTypes.string).isRequired,
  permitGuid: PropTypes.string.isRequired,
  mineGuid: PropTypes.string.isRequired,
  bond: CustomPropTypes.bond,
  editBond: PropTypes.bool,
};

const defaultProps = {
  bond: {},
  editBond: false,
};

export const AddBondModal = (props) => {
  const handleAddBond = (values) =>
    props.editBond
      ? props.onSubmit(values, props.bond.bond_guid)
      : props.onSubmit(values, props.permitGuid);
  const initialPartyValue = props.editBond
    ? {
        key: props.bond.payer_party_guid,
        label: props.bond.payer.name,
      }
    : "";
  return (
    <div>
      {props.editBond && (
        <div>
          <Alert
            message="Making a correction?"
            description="Use this window to make corrections to bonds you have recorded.
            If you want to record a change and keep the history of the bond, release this bond and record a new one instead."
            type="info"
            showIcon
            style={{ textAlign: "left" }}
          />
          <br />
        </div>
      )}
      <BondForm
        onSubmit={handleAddBond}
        closeModal={props.closeModal}
        title={props.title}
        initialPartyValue={initialPartyValue}
        provinceOptions={props.provinceOptions}
        bondTypeDropDownOptions={props.bondTypeDropDownOptions}
        bondDocumentTypeDropDownOptions={props.bondDocumentTypeDropDownOptions}
        bondDocumentTypeOptionsHash={props.bondDocumentTypeOptionsHash}
        initialValues={props.bond}
        bond={props.bond}
        mineGuid={props.mineGuid}
      />
    </div>
  );
};

AddBondModal.propTypes = propTypes;
AddBondModal.defaultProps = defaultProps;

const mapStateToProps = (state) => ({
  provinceOptions: getDropdownProvinceOptions(state),
  bondTypeDropDownOptions: getBondTypeDropDownOptions(state),
  bondDocumentTypeDropDownOptions: getBondDocumentTypeDropDownOptions(state),
  bondDocumentTypeOptionsHash: getBondDocumentTypeOptionsHash(state),
});

export default connect(mapStateToProps)(AddBondModal);
