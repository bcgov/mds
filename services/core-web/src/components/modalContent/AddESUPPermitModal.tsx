import React, { FC } from "react";
import PropTypes from "prop-types";
import CustomPropTypes from "@/customPropTypes";
import AddESUPPermitForm from "@/components/Forms/AddESUPPermitForm";


const propTypes = {};

const defaultProps = {};
const AddESUPPermitModal = (props) => (
  <div>
    <AddESUPPermitForm {...props}/>
    {/* <AddESUPPermitForm
      onSubmit={props.onSubmit}
      closeModal={props.closeModal}
      title={props.title}
      mine_guid={props.mine_guid}
      isPermitTab={props.isPermitTab}
    /> */}
  </div>
);

// AddESUPPermitModal.propTypes = {
//     onSubmit: PropTypes.func.isRequired,
//     title: PropTypes.string.isRequired,
//     initialValues: PropTypes.objectOf(PropTypes.string).isRequired,
//     mineGuid: PropTypes.string.isRequired,
//     isApproved: PropTypes.bool.isRequired,
//     documentTypeDropdownOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
//     isPermitTab: PropTypes.bool.isRequired,
//     inspectors: CustomPropTypes.groupOptions.isRequired,
//   };
AddESUPPermitModal.propTypes = propTypes;
AddESUPPermitModal.defaultProps = defaultProps;

export default AddESUPPermitModal;
