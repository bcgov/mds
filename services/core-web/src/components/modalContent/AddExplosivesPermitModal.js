import React from "react";
import PropTypes from "prop-types";
import CustomPropTypes from "@/customPropTypes";
import ExplosivesPermitForm from "@/components/Forms/ExplosivesPermit/ExplosivesPermitForm";

const propTypes = {
  onSubmit: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.string).isRequired,
  mineGuid: PropTypes.string.isRequired,
  isApproved: PropTypes.bool.isRequired,
  documentTypeDropdownOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
  isPermitTab: PropTypes.bool.isRequired,
  inspectors: CustomPropTypes.groupOptions.isRequired,
  initialMineOperatorValue: PropTypes.objectOf(PropTypes.string).isRequired,
};

export const AddExplosivesPermitModal = (props) => (
  <div>
    <ExplosivesPermitForm {...props} />
  </div>
);

AddExplosivesPermitModal.propTypes = propTypes;

export default AddExplosivesPermitModal;
