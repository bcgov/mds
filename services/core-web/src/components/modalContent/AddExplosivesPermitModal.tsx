import React, { FC } from "react";
import { IOption, IGroupedDropdownList } from "@mds/common";
import ExplosivesPermitForm from "@/components/Forms/ExplosivesPermit/ExplosivesPermitForm";
import { Feature, isFeatureEnabled } from "@mds/common";
import ExplosivesPermitFormNew from "@/components/Forms/ExplosivesPermit/ExplosivesPermitFormNew";

interface ExplosivesPermitModalProps {
  title: string;
  initialValues: any;
  mineGuid: string;
  isApproved: boolean;
  documentTypeDropdownOptions: IOption[];
  isPermitTab: boolean;
  inspectors: IGroupedDropdownList[];
  closeModal: () => void;
  isProcessed: boolean;
  isAmendment: boolean;
}

export const AddExplosivesPermitModal: FC<ExplosivesPermitModalProps> = (props) => (
  <div>
    {isFeatureEnabled(Feature.ESUP_PERMIT_AMENDMENT) ? (
      <ExplosivesPermitFormNew {...props} />
    ) : (
      <ExplosivesPermitForm {...props} />
    )}
  </div>
);

export default AddExplosivesPermitModal;
