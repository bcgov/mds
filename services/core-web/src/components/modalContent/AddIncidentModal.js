// Passing props into function causes linter to not recognize use of props used in that function.
/* eslint-disable react/no-unused-prop-types */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { getFormValues } from "redux-form";
import { concat, reject } from "lodash";
import { Steps, Button, Popconfirm } from "antd";
import * as Strings from "@common/constants/strings";
import * as FORM from "@/constants/forms";
import AddIncidentReportingForm from "@/components/Forms/incidents/AddIncidentReportingForm";
import AddIncidentDetailForm from "@/components/Forms/incidents/AddIncidentDetailForm";
import AddIncidentFollowUpForm from "@/components/Forms/incidents/AddIncidentFollowUpForm";

import CustomPropTypes from "@/customPropTypes";

const propTypes = {
  newIncident: PropTypes.bool.isRequired,
  onSubmit: PropTypes.func.isRequired,
  afterClose: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  incidentDeterminationOptions: CustomPropTypes.options.isRequired,
  incidentStatusCodeOptions: CustomPropTypes.options.isRequired,
  incidentCategoryCodeOptions: CustomPropTypes.options.isRequired,
  followupActionOptions: CustomPropTypes.options.isRequired,
  doSubparagraphOptions: CustomPropTypes.options.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
  inspectors: CustomPropTypes.groupOptions.isRequired,
  addIncidentFormValues: PropTypes.objectOf(PropTypes.any),
  mineGuid: PropTypes.string.isRequired,
};

const defaultProps = {
  addIncidentFormValues: {},
};

const invalidReportingPayload = (values) =>
  !(
    values.reported_date &&
    values.reported_time &&
    values.reported_by_name &&
    values.reported_to_inspector_party_guid &&
    values.responsible_inspector_party_guid &&
    values.categories &&
    values.categories.length > 0
  );

const invalidDetailPayload = (values) =>
  !(
    values.incident_date &&
    values.incident_time &&
    values.incident_description &&
    values.determination_type_code &&
    // If DO, need subparagraphs
    ((values.determination_type_code === Strings.INCIDENT_DETERMINATION_TYPES.dangerousOccurance &&
      values.determination_inspector_party_guid &&
      values.dangerous_occurrence_subparagraph_ids &&
      values.dangerous_occurrence_subparagraph_ids.length !== 0) ||
      (values.determination_type_code ===
        Strings.INCIDENT_DETERMINATION_TYPES.notADangerousOccurance &&
        values.status_code &&
        values.determination_inspector_party_guid) ||
      values.determination_type_code === Strings.INCIDENT_DETERMINATION_TYPES.pending)
  );

const actionVerb = (newIncident) => {
  if (newIncident) return <span>Save&nbsp;</span>;
  return <span>Edit&nbsp;</span>;
};

const StepForms = (
  props,
  state,
  next,
  prev,
  handleIncidentSubmit,
  uploadedFiles,
  onFileLoad,
  onRemoveFile,
  invalidFollowUpPayload
) => [
  {
    title: "Initial Report",
    content: (
      <AddIncidentReportingForm
        initialValues={props.initialValues}
        inspectors={props.inspectors}
        incidentCategoryCodeOptions={props.incidentCategoryCodeOptions}
      />
    ),
    buttons: (
      <Button
        id="step1-next"
        type="tertiary"
        className="full-mobile"
        onClick={() => next()}
        disabled={state.submitting || invalidReportingPayload(props.addIncidentFormValues)}
      >
        Next
      </Button>
    ),
  },
  {
    title: "Add Details",
    content: (
      <AddIncidentDetailForm
        mineGuid={props.mineGuid}
        initialValues={props.initialValues}
        doSubparagraphOptions={props.doSubparagraphOptions}
        incidentDeterminationOptions={props.incidentDeterminationOptions}
        incidentStatusCodeOptions={props.incidentStatusCodeOptions}
        inspectors={props.inspectors}
        doDetermination={props.addIncidentFormValues.determination_type_code}
        uploadedFiles={uploadedFiles.filter(
          (file) =>
            file.mine_incident_document_type_code === Strings.INCIDENT_DOCUMENT_TYPES.initial
        )}
        onFileLoad={onFileLoad}
        onRemoveFile={onRemoveFile}
      />
    ),
    buttons: (
      <span>
        <Button
          id="step-back"
          type="tertiary"
          className="full-mobile"
          onClick={() => prev()}
          disabled={state.submitting}
        >
          Back
        </Button>
        {props.addIncidentFormValues.determination_type_code !==
          Strings.INCIDENT_DETERMINATION_TYPES.notADangerousOccurance && (
          <Button
            id="step2-next"
            type="tertiary"
            className="full-mobile"
            onClick={() => next()}
            disabled={state.submitting || invalidDetailPayload(props.addIncidentFormValues)}
          >
            Next
          </Button>
        )}
        <Button
          type="primary"
          className="full-mobile"
          onClick={(event) => handleIncidentSubmit(event, false)}
          disabled={invalidDetailPayload(props.addIncidentFormValues)}
          loading={state.submitting}
        >
          {actionVerb(props.newIncident)}
          {props.addIncidentFormValues.determination_type_code !==
            Strings.INCIDENT_DETERMINATION_TYPES.notADangerousOccurance && (
            <span>Initial&nbsp;</span>
          )}
          Incident
        </Button>
      </span>
    ),
  },
  {
    title: "Follow Up",
    content: (
      <AddIncidentFollowUpForm
        mineGuid={props.mineGuid}
        initialValues={props.initialValues}
        incidentDeterminationOptions={props.incidentDeterminationOptions}
        followupActionOptions={props.followupActionOptions}
        incidentStatusCodeOptions={props.incidentStatusCodeOptions}
        hasFatalities={props.addIncidentFormValues.number_of_fatalities > 0 || false}
        hasFollowUp={props.addIncidentFormValues.followup_inspection || false}
        determinationTypeCode={props.addIncidentFormValues.determination_type_code}
        uploadedFiles={uploadedFiles.filter(
          (file) => file.mine_incident_document_type_code === Strings.INCIDENT_DOCUMENT_TYPES.final
        )}
        onFileLoad={onFileLoad}
        onRemoveFile={onRemoveFile}
      />
    ),
    buttons: [
      <Button
        id="step-back"
        type="tertiary"
        className="full-mobile"
        onClick={() => prev()}
        disabled={state.submitting}
      >
        Back
      </Button>,
      <Button
        type="primary"
        className="full-mobile"
        onClick={(event) => handleIncidentSubmit(event, false)}
        disabled={invalidFollowUpPayload(props.addIncidentFormValues)}
        loading={state.submitting}
      >
        {actionVerb(props.newIncident)}
        Incident
      </Button>,
    ],
  },
];

export class AddIncidentModal extends Component {
  state = {
    current: 0,
    uploadedFiles: this.props.initialValues.documents
      ? [...this.props.initialValues.documents]
      : [],
    submitting: false,
  };

  formatTimestamp = (dateString, momentInstance) =>
    dateString && momentInstance && `${dateString} ${momentInstance.format("HH:mm")}`;

  parseFormDataIntoPayload = ({
    reported_date,
    reported_time,
    incident_date,
    incident_time,
    ...remainingValues
  }) => ({
    ...remainingValues,
    reported_timestamp: this.formatTimestamp(reported_date, reported_time),
    incident_timestamp: this.formatTimestamp(incident_date, incident_time),
  });

  handleIncidentSubmit = () => {
    this.setState({ submitting: true });
    this.props
      .onSubmit({
        ...this.parseFormDataIntoPayload(this.props.addIncidentFormValues),
        updated_documents: this.state.uploadedFiles,
      })
      .then(() => this.close())
      .finally(() => this.setState({ submitting: false }));
  };

  close = () => {
    this.props.closeModal();
    this.props.afterClose();
  };

  next = () => this.setState((prevState) => ({ current: prevState.current + 1 }));

  prev = () => this.setState((prevState) => ({ current: prevState.current - 1 }));

  onFileLoad = (document_name, document_manager_guid, mine_incident_document_type_code) =>
    this.setState((prevState) => ({
      uploadedFiles: concat(prevState.uploadedFiles, {
        document_name,
        document_manager_guid,
        mine_incident_document_type_code,
      }),
    }));

  onRemoveFile = (file) => {
    this.setState((prevState) => ({
      uploadedFiles: reject(
        prevState.uploadedFiles,
        (uploadedFile) => file.document_manager_guid === uploadedFile.document_manager_guid
      ),
    }));
  };

  invalidFollowUpPayload = (values) => {
    let disableSubmit = true;
    const finalDocs = this.state.uploadedFiles.filter(
      (file) => file.mine_incident_document_type_code === Strings.INCIDENT_DOCUMENT_TYPES.final
    );
    if (values.status_code && values.followup_investigation_type_code) {
      if (finalDocs.length > 0) {
        disableSubmit = false;
      } else if (finalDocs.length === 0 && values.status_code === "PRE") {
        disableSubmit = false;
      }
    }
    return disableSubmit;
  };

  render = () => {
    const Forms = StepForms(
      this.props,
      this.state,
      this.next,
      this.prev,
      this.handleIncidentSubmit,
      this.state.uploadedFiles,
      this.onFileLoad,
      this.onRemoveFile,
      this.invalidFollowUpPayload
    );

    return (
      <div>
        <div>
          <div>
            <Steps current={this.state.current}>
              {Forms.map((step) => (
                <Steps.Step key={step.title} title={step.title} />
              ))}
            </Steps>
            <br />

            <div>{Forms[this.state.current].content}</div>

            <div className="right center-mobile">
              <Popconfirm
                placement="top"
                title="Are you sure you want to cancel?"
                okText="Yes"
                cancelText="No"
                onConfirm={this.close}
                disabled={this.state.submitting}
              >
                <Button type="secondary" className="full-mobile" disabled={this.state.submitting}>
                  Cancel
                </Button>
              </Popconfirm>

              {Forms[this.state.current].buttons}
            </div>
          </div>
        </div>
      </div>
    );
  };
}

const mapStateToProps = (state) => ({
  addIncidentFormValues: getFormValues(FORM.MINE_INCIDENT)(state) || {},
});

AddIncidentModal.propTypes = propTypes;
AddIncidentModal.defaultProps = defaultProps;

export default connect(mapStateToProps)(AddIncidentModal);
