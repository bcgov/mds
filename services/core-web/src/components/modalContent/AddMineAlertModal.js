import React from "react";
import PropTypes from "prop-types";
import AddMineAlertForm from "@/components/Forms/mineAlerts/AddMineAlertForm";
import customPropTypes from "@/customPropTypes";

const propTypes = {
  onSubmit: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  mineGuid: PropTypes.string.isRequired,
  mine: customPropTypes.mine.isRequired,
};

export const AddMineAlertModal = (props) => (
  <div>
    <AddMineAlertForm
      onSubmit={props.onSubmit}
      closeModal={props.closeModal}
      mine={props.mine}
      title={props.title}
      {...props}
    />
  </div>
);

AddMineAlertModal.propTypes = propTypes;

export default AddMineAlertModal;
