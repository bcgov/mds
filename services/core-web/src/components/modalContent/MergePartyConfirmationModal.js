import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import MergePartyConfirmationForm from "@/components/Forms/parties/MergePartyConfirmationForm";
import { getDropdownProvinceOptions } from "@mds/common/redux/selectors/staticContentSelectors";
import CustomPropTypes from "@/customPropTypes";

const propTypes = {
  closeModal: PropTypes.func.isRequired,
  initialValues: PropTypes.objectOf(PropTypes.any).isRequired,
  provinceOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
  title: PropTypes.string.isRequired,
  isPerson: PropTypes.bool.isRequired,
  partyRelationshipTypesHash: PropTypes.objectOf(PropTypes.string).isRequired,
  roles: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
};

const defaultProps = {};

export const MergePartyConfirmationModal = (props) => {
  return (
    <div>
      <MergePartyConfirmationForm
        isPerson={props.isPerson}
        initialValues={props.initialValues}
        provinceOptions={props.provinceOptions}
        closeModal={props.closeModal}
        title={props.title}
        roles={props.roles}
        partyRelationshipTypesHash={props.partyRelationshipTypesHash}
        {...props}
      />
    </div>
  );
};

const mapStateToProps = (state) => ({
  provinceOptions: getDropdownProvinceOptions(state),
});

MergePartyConfirmationModal.propTypes = propTypes;
MergePartyConfirmationModal.defaultProps = defaultProps;

export default connect(mapStateToProps)(MergePartyConfirmationModal);
