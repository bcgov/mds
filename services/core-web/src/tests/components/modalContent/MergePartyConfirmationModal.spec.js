import React from "react";
import { shallow } from "enzyme";
import { MergePartyConfirmationModal } from "@/components/modalContent/MergePartyConfirmationModal";

const dispatchProps = {};
const props = {};

const setupDispatchProps = () => {
  dispatchProps.onSubmit = jest.fn();
  dispatchProps.closeModal = jest.fn();
};

const setupProps = () => {
  props.title = "mockTitle";
  props.initialValues = {};
  props.provinceOptions = [];
  props.isPerson = false;
  props.partyRelationshipTypesHash = {};
  props.roles = [];
};

beforeEach(() => {
  setupDispatchProps();
  setupProps();
});

describe("MergePartyConfirmationModal", () => {
  it("renders properly", () => {
    const component = shallow(<MergePartyConfirmationModal {...dispatchProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
});
