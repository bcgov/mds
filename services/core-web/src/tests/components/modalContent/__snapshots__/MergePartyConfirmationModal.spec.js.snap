// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MergePartyConfirmationModal renders properly 1`] = `
<div>
  <ReduxForm
    closeModal={[MockFunction]}
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="MERGE_PARTY_CONFIRMATION"
    getFormState={[Function]}
    initialValues={Object {}}
    isPerson={false}
    keepDirtyOnReinitialize={false}
    onSubmit={[MockFunction]}
    onSubmitSuccess={[Function]}
    partyRelationshipTypesHash={Object {}}
    persistentSubmitErrors={false}
    provinceOptions={Array []}
    pure={true}
    roles={Array []}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    title="mockTitle"
    touchOnBlur={false}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />
</div>
`;
